app/routes/tags/components/TagDetail.tsx

```diff
@@ -31,9 +32,14 @@ const link = (key: string, tag: string) => {
   return links[key] || <h4>{translate(key)}</h4>;
 };
 
+type TagDetailParams = {
+  tagId: string;
+};
 const TagDetail = () => {
-  const { tagId } = useParams();
-  const tag = useAppSelector((state) => selectTagById(state, { tagId }));
+  const { tagId } = useParams<TagDetailParams>() as TagDetailParams;
+  const tag = useAppSelector((state) =>
+    selectTagById(state, tagId),
+  ) as DetailedTag;
```

Comment:
Can't this be `undefined` as well? Or an empty object?

Reply:
Yup, it can. I'll fix:)